### PR TITLE
docs: add dark mode

### PR DESCRIPTION
**Pull request by kyletsang**
Adds dark mode to the site.

- Fixes examples so they look better in dark mode
- Fix random doc issues I found along the way
- Add docs regarding color mode
- Add deprecation warnings for components with dark classes

Preview: https://deploy-preview-6633--react-bootstrap.netlify.app/

### CODE REVIEW

File: www/docs/components/close-button.mdx [hunk 1]
```diff
@@ -27,6 +27,14 @@ interactions.
 
 Change the default dark color to white using `variant="white"`.
 
+:::caution Heads up!
+
+Dark variants for components were deprecated in Bootstrap v5.3.0 with the introduction
+of color modes. Instead of adding `variant="white"`, set `data-bs-theme="dark"` on the root
```
Comment (golota60) [inline www/docs/components/close-button.mdx:33]: i think you might've meant `data-bs-theme="light"` here

File: www/docs/examples/Accordion/ContextAwareToggle.js [hunk 2]
```diff
@@ -17,7 +17,7 @@ function ContextAwareToggle({ children, eventKey, callback }) {
   return (
     <button
       type="button"
-      style={{ backgroundColor: isCurrentEventKey ? 'pink' : 'lavender' }}
+      style={{ backgroundColor: isCurrentEventKey ? 'rgba(255,192,203,0.6)' : 'rgb(230,230,250,0.6)' }}
```
Comment (golota60) [inline www/docs/examples/Accordion/ContextAwareToggle.js:20]: [nitpick] - those could be moved to variables describing what colors they are

**Review by golota60 — APPROVED** [review]
lgtm 👍

**Review by jquense — APPROVED** [review]
Nice

**Pull request merged**